### PR TITLE
Fix F841 flake8 warning

### PR DESCRIPTION
**Pull request by aaugustin**
This PR completes the "fix all flake8 warnings" effort of the last few days.

There's a lot of judgment in the fixes here: when the result of an expression is assigned to a variable that isn't used:

- if the expression has no side effect, then it can safely be removed
- if the expression has side effects, then it must be kept and only the assignment to a variable must be removed
- or it may be a coding / refactoring mistake that results in a badly named variable

I'm not sure I made the right call in all cases, so I would appreciate a review.

E203, E501, W503 are still ignored because they're debatable, black disagrees with flake8, and black wins (by not being configurable).

### CODE REVIEW

File: examples/contrib/run_openai_gpt.py [hunk 1]
```diff
@@ -182,9 +179,6 @@ def main():
     model.to(device)
 
     # Load and encode the datasets
-    if not args.train_dataset and not args.eval_dataset:
-        roc_stories = cached_path(ROCSTORIES_URL)
```
Comment (aaugustin) [inline examples/contrib/run_openai_gpt.py]: I fail to see how the example uses this. I'm not seeing any code turning `roc_stories` into datasets.

File: examples/contrib/run_transfo_xl.py [hunk 2]
```diff
@@ -73,9 +73,7 @@ def main():
     # The pre-processing involve computing word frequencies to prepare the Adaptive input and SoftMax
     # and tokenizing the dataset
     # The pre-processed corpus is a convertion (using the conversion script )
-    tokenizer = TransfoXLTokenizer.from_pretrained(args.model_name)
```
Comment (aaugustin) [inline examples/contrib/run_transfo_xl.py]: Looking at the implementation `TransfoXLTokenizer` and its parent `PreTrainedTokenizer`, this doesn't set any global state, so it can be safely removed.

File: src/transformers/modeling_t5.py [hunk 3]
```diff
@@ -629,7 +629,6 @@ def forward(
                 all_attentions = all_attentions + (layer_outputs[1],)  # We keep only self-attention weights for now
 
         hidden_states = self.final_layer_norm(hidden_states)
-        layer_output = self.dropout(hidden_states)
```
Comment (aaugustin) [inline src/transformers/modeling_t5.py]: I'm not sufficiently familiar with PyTorch to be sure this has no side effects.

I believe the intent is to generate an output layer by applying dropout to the hidden states, and since the output layer isn't actually used, it's safe to remove this operation.

File: src/transformers/modeling_tf_pytorch_utils.py [hunk 4]
```diff
@@ -122,7 +122,7 @@ def load_pytorch_weights_in_tf2_model(tf_model, pt_state_dict, tf_inputs=None, a
         tf_inputs = tf_model.dummy_inputs
 
     if tf_inputs is not None:
-        tfo = tf_model(tf_inputs, training=False)  # Make sure model is built
+        tf_model(tf_inputs, training=False)  # Make sure model is built
```
Comment (aaugustin) [inline src/transformers/modeling_tf_pytorch_utils.py]: I kept this (and all similar cases) because the comment suggests it's an in-place operation.

File: src/transformers/modeling_tf_t5.py [hunk 5]
```diff
@@ -491,7 +491,6 @@ def call(
                 all_attentions = all_attentions + (layer_outputs[1],)
 
         hidden_states = self.final_layer_norm(hidden_states)
-        layer_output = self.dropout(hidden_states, training=training)
```
Comment (aaugustin) [inline src/transformers/modeling_tf_t5.py]: Same comment as in `modeling_t5.py`.

File: src/transformers/modeling_tf_utils.py [hunk 6]
```diff
@@ -515,7 +515,7 @@ def call(self, inputs, training=False):
             cls_index = inputs[1] if len(inputs) > 1 else None
             assert len(inputs) <= 2, "Too many inputs."
         else:
-            input_ids = inputs.get("input_ids")
+            hidden_states = inputs.get("hidden_states")
```
Comment (aaugustin) [inline src/transformers/modeling_tf_utils.py]: I made this change because:

- the docstring suggests the input argument should contain hidden states; and
- without this change, if the code goes through this `else` branch, it'll fail soon thereafter because the `hidden_states` variable isn't defined.

This is a wild guess.

File: tests/test_modeling_common.py [hunk 7]
```diff
@@ -590,7 +589,7 @@ def test_inputs_embeds(self):
                 inputs_dict["decoder_inputs_embeds"] = wte(decoder_input_ids)
 
             with torch.no_grad():
-                outputs = model(**inputs_dict)
+                model(**inputs_dict)
```
Comment (aaugustin) [inline tests/test_modeling_common.py]: I kept this, and all similar cases, because I assume we're testing that the model runs without crashing.

File: tests/test_modeling_tf_xlm.py [hunk 8]
```diff
@@ -224,7 +224,6 @@ def create_and_check_xlm_qa(
 
             inputs = {"input_ids": input_ids, "lengths": input_lengths}
 
-            outputs = model(inputs)
```
Comment (aaugustin) [inline tests/test_modeling_tf_xlm.py]: I removed this because it looks like a duplicate with the next line.

File: templates/adding_a_new_example_script/utils_xxx.py [hunk 9]
```diff
@@ -868,8 +868,6 @@ def write_predictions_extended(
         orig_data = json.load(reader)["data"]
 
     qid_to_has_ans = make_qid_to_has_ans(orig_data)
-    has_ans_qids = [k for k, v in qid_to_has_ans.items() if v]
-    no_ans_qids = [k for k, v in qid_to_has_ans.items() if not v]
```
Comment (aaugustin) [inline templates/adding_a_new_example_script/utils_xxx.py]: I'm not seeing any example scripts containing similar code so it doesn't look useful to keep in the template.

File: src/transformers/modeling_albert.py [hunk 10]
```diff
@@ -334,9 +332,6 @@ def forward(self, hidden_states, attention_mask=None, head_mask=None):
             # Index of the hidden group
             group_idx = int(i / (self.config.num_hidden_layers / self.config.num_hidden_groups))
 
-            # Index of the layer inside the group
-            layer_idx = int(i - group_idx * layers_per_group)
```
Comment (thomwolf) [inline src/transformers/modeling_albert.py]: cc @LysandreJik please check that this `layer_idx` was indeed dead-code and something that should have been used in the subsequent lines

File: src/transformers/modeling_t5.py (hunk 3 above)
Comment (thomwolf) [reply src/transformers/modeling_t5.py]: This dropout should actually be used (cf. https://github.com/tensorflow/mesh/blob/master/mesh_tensorflow/transformer/transformer.py#L340)
It has a very small effect though but I'll add it for correctness.

**Review by thomwolf — APPROVED** [review]
